### PR TITLE
Application code refactoring (EA-159)

### PR DESCRIPTION
**Pull request by laleks6**
!!! **Pay attention, that we don't merge pull-requests without teammembers approval** !!!
!!! **At least one of us must check your changes.** !!!

## Proposed changes

- Describe the big picture of your changes here:
 Application code refactoring
Change link footer

## Types of changes

What types of changes does your code introduce to eCommerce-application?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added necessary documentation (if appropriate)

## Related tickets & documents

- Link to the ticket: [link](https://kategoncharik.atlassian.net/jira/software/projects/EA/boards/1?selectedIssue=EA-159)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/app/sdk/requests.ts [hunk 1]
```diff
@@ -17,24 +13,29 @@ import {
...
+import { markInputAsInvalid } from '@helpers/toggle-validation-state';
 import { ProductData } from '@customTypes/data-product';
+import { DataUser } from '@customTypes/datauser';
+import { rememberAuthorizedUser } from '@app/state';
 
 const errorMessage = 'Error: no connection to server';
 export const productsPerPage = 12;
 
-export async function createUser(form: DataUser): Promise<number | undefined> {
+export async function createUser(form: DataUser): Promise<number | undefined | null> {
```
Comment (Irina-Grebennikova) [inline src/app/sdk/requests.ts]: здесь надо сделать, чтобы функция возвращала либо number, либо null

File: src/app/sdk/requests.ts [hunk 2]
```diff
@@ -43,10 +44,11 @@ export async function isUserExist(email: string): Promise<boolean | void> {
     return result.body.count > 0;
   } catch (err) {
     console.error(errorMessage);
+    return null;
   }
 }
 
-export async function authorizeUser(email: string, password: string): Promise<void | string> {
+export async function authorizeUser(email: string, password: string): Promise<void | string | null> {
```
Comment (Irina-Grebennikova) [inline src/app/sdk/requests.ts]: раз уж мы меняем везде возврат void на null, надо здесь также сделать, чтоб без void было

File: src/app/sdk/requests.ts [hunk 3]
```diff
@@ -139,10 +142,13 @@ export const getProduct = async (key: string): Promise<ProductData | void> => {
...
     })
-    .catch(() => console.error(errorMessage));
+    .catch(() => {
+      console.error(errorMessage);
+      return null;
+    });
 };
 
-export async function getProductsOfCategory(id: string, offset = 0): Promise<ProductProjection[]> {
+export async function getProductsOfCategory(id: string, offset = 0): Promise<ProductProjection[] | null> {
```
Comment (Irina-Grebennikova) [inline src/app/sdk/requests.ts]: в данном случае, мне кажется, не обязательно менять было

File: src/app/sdk/requests.ts (hunk 1 above)
Comment (laleks6) [reply src/app/sdk/requests.ts]: исправил

File: src/app/sdk/requests.ts (hunk 2 above)
Comment (laleks6) [reply src/app/sdk/requests.ts]: исправид

File: src/app/sdk/requests.ts (hunk 3 above)
Comment (laleks6) [reply src/app/sdk/requests.ts]: исправил

**Review by Irina-Grebennikova — APPROVED** [review]

**Pull request merged**